File: android/app/src/main/java/com/blackatclient/SignalModule.java
package com.blackatclient;

import android.os.AsyncTask;
import android.util.Log;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.WritableArray;
import com.facebook.react.bridge.WritableNativeArray;

import org.signal.libsignal.protocol.IdentityKey;
import org.signal.libsignal.protocol.IdentityKeyPair;
import org.signal.libsignal.protocol.ecc.ECPrivateKey;

public class SignalModule extends ReactContextBaseJavaModule {
    SignalModule(ReactApplicationContext context) {
        super(context);
    }
    @NonNull
    @Override
    public String getName() {
        return "SignalModule";
    }

    @ReactMethod
    public void generateIdentityKeyPair(Promise promise) {
        IdentityKeyPair identityKeyPair = IdentityKeyPair.generate();
        promise.resolve(Utils.fromBytes(identityKeyPair.serialize()));
    }
}

File: android/app/src/main/java/com/blackat/chat/module/SignalModule.kt
package com.blackat.chat.module

import com.blackatclient.Utils
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import org.signal.libsignal.protocol.IdentityKeyPair

class SignalModule internal constructor(context: ReactApplicationContext?) : ReactContextBaseJavaModule(context) {
    override fun getName(): String {
        return "SignalModule"
    }

    @ReactMethod
    fun generateIdentityKeyPair(promise: Promise) {
        val identityKeyPair = IdentityKeyPair.generate()
        promise.resolve(Utils.fromBytes(identityKeyPair.serialize()))
    }
}